### ISSUE TITLE
Critical Crash in iOS

### ISSUE THREAD

**Issue by luis901101** [juliuscanute/qr_code_scanner#56]
I have been using this plugin for a while now and it was fine, but recently in the latest iOS versions (13.1.3 onwards) my app closes unexpectedly while scanning... It is a crash in some native level because I have implemented Crash Reports in my app and it doesn't get noticed... the app just closes itself. It is not a constant thing I mean sometimes I can scan 100 codes but it crashes at 101, other times it crashes at 5 code scans... it is something random but always crashes for sure.
In order to discard any bug related to my app I created an example project to test specifically this plugin v0.0.12 and it crashes in the same way.

**Note 0: in the _onQRViewCreated in this example I use pauseCamera() and resumeCamera() with a Future.delay() to simulate an API request delay but I also tested without this pause/resume delay and it crashes the same**

**Note 1: I have tested in iPhone6S with iOS 13.2, iPhone 6S Plus with iOS 13.1.3, iPhone X with iOS 13.1.3 and iPad 6th Generation with iOS 13.2**

**Note 2: In Android it seems to work fine the app doesn't crash.**

I used the exact main.dart for the example of this plugin and just added a scansCount to know when exactly it crashes... 
**[Edited main with sync and async scans simulation]**
[main.dart.txt](https://github.com/juliuscanute/qr_code_scanner/files/3820395/main.dart.txt)

**Comment by luis901101** [comment]
This is my flutter doctor output:

[✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.15 19A583, locale en-US)
    • Flutter version 1.9.1+hotfix.6 at /Users/krrigan/Flutter/SDK
    • Framework revision 68587a0916 (8 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/krrigan/Android/SDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/krrigan/Android/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2, Build version 11B52
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 23.2.1
    • Dart plugin version 171.4424
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[!] VS Code (version 1.37.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • MyIphone • asdasdasdasd • ios • iOS 13.2

! Doctor found issues in 1 category.

**Comment by luis901101** [comment]
Recently I found that after some sync scans _(by sync scans I refer to a scan followed by a pasueCamera followed by a delay followed by a resumeCamera... you can see it here https://github.com/juliuscanute/qr_code_scanner/files/3820395/main.dart.txt)_ the interface freezes for a while... you can see the video below... but the most important thing still being the random app crash...

https://www.dropbox.com/s/uw0elp6li60g4yb/qrcodescanner-example-screen_freeze.mp4?raw=1

**Comment by kamilpowalowski** [comment]
Since you didn't provide native crash log I'm not sure, but it may be related to the crash that I found and fixed in https://github.com/juliuscanute/qr_code_scanner/pull/69

**Comment by luis901101** [comment]
I will run the app from XCode to try to get the native log but in the mean time I would like to try your fix... Is this fix going to be available in v0.0.13 of the plugin and when? Or do I need to clone and test it as a local dependency?

**Comment by kamilpowalowski** [comment]
You can modify your Pubspec.yaml to take my branch for testing.
```yaml
qr_code_scanner:
    git:
      url: https://github.com/kamilpowalowski/qr_code_scanner.git
      ref: fix/ios-crash-on-not-correct-code
```

When it will be merged it depends to @juliuscanute

**Comment by luis901101** [comment]
Ok thanks. @juliuscanute it's away for a long time, I will use your fix by the moment

**Comment by asada03** [comment]
I am having a similar problem. The plugin works great on Android, but when used on iOS it crashes.

I am testing on an iPhone XR with ios 13.1.2.

For me, it crashes every time. As soon as the QRView widget is added the app closes with no error log. The only thing I see in the error log is "Lost connection to device".

@kamilpowalowski I have tried using your branch too, with the same results.

Thanks

**Comment by shinsenter** [comment]
@juliuscanute 
This may also fix #75. Please merge!

**Comment by kamilpowalowski** [comment]
@shinsenter Correct #75 is the error that I had and I fixed it in https://github.com/juliuscanute/qr_code_scanner/pull/69

**Comment by dspicher** [comment]
This should be fixed thanks to #72 that fixed the same issue as #69.
